universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -21,10 +22,14 @@
 public final class QuestionServiceImpl implements QuestionService {
 
   private QuestionRepository questionRepository;
+  private Provider<VersionRepository> versionRepositoryProvider;
```

Comment:
Suggestion (non-blocking): make this and both `private final`

Reply:
Done.